### PR TITLE
Add .domains and .enclaves on GatewayClient as additional properties

### PR DESCRIPTION
**Pull request by yashgorana**
## Description

Closes https://github.com/OpenMined/Heartbeat/issues/526

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by yashgorana** [comment]
TODO: Unit tests for this change. Please don't merge until i have some unit tests added to this change.

**Comment by yashgorana** [comment]
@rasswanth-s @shubham3121 have pushed all the unit test changes and PR is ready to be merged.

File: packages/syft/src/syft/client/gateway_client.py [hunk 1]
```diff
@@ -55,3 +56,11 @@ def proxy_client_for(
         if email and password:
             res.login(email=email, password=password, **kwargs)
         return res
+
+    @property
+    def domains(self):
+        return self.api.services.network.get_peers_by_type(node_type=NodeType.DOMAIN)
```
Comment (shubham3121) [inline packages/syft/src/syft/client/gateway_client.py:67]: Can we also add the check for the network service:
if self.api is not None and self.api.has_service("network"):

File: packages/syft/src/syft/client/gateway_client.py [hunk 2]
```diff
@@ -55,3 +56,11 @@ def proxy_client_for(
...
+
+    @property
+    def enclaves(self):
+        return self.api.services.network.get_peers_by_type(node_type=NodeType.ENCLAVE)
```
Comment (shubham3121) [inline packages/syft/src/syft/client/gateway_client.py:73]: Similarly here:
if self.api is not None and self.api.has_service("network"):

**Review by shubham3121 — APPROVED** [review]
Awesome work @yashgorana !! 

Left one minor comment, otherwise looks good.

File: packages/syft/src/syft/client/gateway_client.py (hunk 1 above)
Comment (yashgorana) [reply packages/syft/src/syft/client/gateway_client.py:67]: Ah good catch. Actually, we don't need `self.api is not None` as the property checks None and fetches + returns the API for us anyway.

**Review by rasswanth-s — APPROVED** [review]
Well Done @yashgorana , the changes looks great

**Review by shubham3121 — APPROVED** [review]

**Pull request merged**